src/utils/validator.ts

```diff
@@ -143,187 +144,199 @@ export function validateNonce(
 	return nonce
 }
 
+/**
+ * Validates a numeric ID (e.g., chain_id, vault_id)
+ */
+function validateId(id: unknown, fieldName: string): number {
+	if (typeof id !== 'number') {
+		throw new ValidationError('ID must be a number', fieldName, id, {
+			receivedType: typeof id,
+		})
+	}
+
+	if (!Number.isInteger(id)) {
+		throw new ValidationError('ID must be an integer', fieldName, id)
+	}
+
+	if (id < 0) {
+		throw new ValidationError('ID must not be negative', fieldName, id)
+	}
+
+	return id
+}
+
 /**
  * Validates an intention object
  */
 export function validateIntention(intention: Intention): Intention {
 	const startTime = Date.now()
-	const validated: Intention = { ...intention }
 
-	try {
-		// Validate addresses if present
-		if (intention.from) {
-			validated.from = validateAddress(intention.from, 'intention.from')
-		}
-		if (intention.to) {
-			validated.to = validateAddress(intention.to, 'intention.to')
-		}
-		if (intention.from_token_address) {
-			validated.from_token_address = validateAddress(
-				intention.from_token_address,
-				'intention.from_token_address'
-			)
-		}
-		if (intention.to_token_address) {
-			validated.to_token_address = validateAddress(
-				intention.to_token_address,
-				'intention.to_token_address'
-			)
-		}
+	if (typeof intention.action !== 'string' || intention.action === '') {
+		throw new ValidationError(
+			'Action is required and must be a non-empty string',
+			'intention.action',
+			intention.action
+		)
+	}
 
-		// Validate amounts if present
-		if (intention.amount_sent) {
-			validated.amount_sent = validateBalance(
-				intention.amount_sent,
-				'intention.amount_sent'
-			)
+	const validated: Intention = {
```

Comment:
Related to a note above: Let's rename `tip` to `proposerTip`, add an additional fee field for `agentTip`, and also add an `expiry` field, which will be a future timestamp when the intention is no longer valid.